### DISCUSSION TITLE
How can Rect be combined with Text to become a group, changing the size will not cause Text deformation?

### DISCUSSION THREAD

**Discussion by Lin-Min** [fabricjs/fabric.js#10305] [Q&A]
How can Rect and Text be combined into a group so that the text content does not deform when the group shape is changed? Using version 6.5.1, my code is as follows：
`
      
      const textbox = new fabric.Textbox('GZWC-B-23F-B-006', {
        width:22,
        fontSize: 4,
        lineHeight:1,
        originX: 'center',
        originY: 'center',
        splitByGrapheme: true,
      })

      const rect = new fabric.Rect({
        width: 22,
        height: 30,
        opacity:.8,
        originX: 'center',
        originY: 'center',
        fill:SPACE_TYPE_COLOR[type],
      })
      
      const group = new fabric.Group([rect, textbox])

      canvas.add(group);
      canvas.requestRenderAll();
`

**Comment by asturur** [comment]
This is a custom functionality that you have to build yourself.
What are you trying to do? do you need a color background for text or something else?

**Reply by Lin-Min** [reply]
I need to add text content to the Rect, and the text content is within the Rect width and height range, and changing the Rect size will not deform the text content.

**Reply by Lin-Min** [reply]
I'm using version 6.5.1

**Reply by Lin-Min** [reply]
Looking at the examples on the website, all I can think of is group to combine them

**Reply by asturur** [reply]
You can group them but you need to write custom controls and custom layout manager for the group.
Is unclear to me if you only need a textbox that render differently, and so you should go on the custom object route.
Both approaches require a considerable amount of custom code and tries

**Reply by Lin-Min** [reply]
> You can group them but you need to write custom controls and custom layout manager for the group. Is unclear to me if you only need a textbox that render differently, and so you should go on the custom object route. Both approaches require a considerable amount of custom code and tries

Do not understand your scheme, can you provide a reference example?

**Reply by Lin-Min** [reply]
Do not understand your scheme, can you provide a reference example?

**Comment by Lin-Min** [comment]
I need to add text content to the Rect that falls within the width and height of the Rect, and that does not deform the text content with any width or height of the Rect.

**Comment by Lin-Min** [comment]
I'm using version 6.5.1.
Looking at the examples on the website, all I can think of is group to combine them.

**Comment by Lin-Min** [comment]
Do not understand your scheme, can you provide a reference example?
